scripts/katana_version/__init__.py

```diff
@@ -68,72 +68,88 @@ def __init__(self, args=Namespace()):
-                self.upstream_remote = "origin"
-                self.origin_remote = "origin"
-            self.upstream_url = git.get_remote_url(self.upstream_remote, self.katana_repo_path)
-            self.origin_url = git.get_remote_url(self.origin_remote, self.katana_repo_path)
-
-            if self.has_enterprise:
-                enterprise_remotes = git.get_remotes(self.katana_enterprise_repo_path)
-                if self.origin_remote not in enterprise_remotes or self.upstream_remote not in enterprise_remotes:
-                    raise ConfigurationError("Missing remotes in enterprise, they should match open.")
-                url = git.get_remote_url(self.upstream_remote, self.katana_enterprise_repo_path)
-                if url.username != self.upstream_url.username:
-                    raise ConfigurationError("Upstream remotes are not in the same user")
-
-                self.enterprise_upstream_url = git.get_remote_url(
-                    self.upstream_remote, self.katana_enterprise_repo_path
-                )
-                self.enterprise_origin_url = git.get_remote_url(self.origin_remote, self.katana_enterprise_repo_path)
-            else:
-                self.enterprise_upstream_url = None
-                self.enterprise_origin_url = None
+            self.open = self._find_katana_remotes(katana_repo_path)
+
+            if katana_enterprise_repo_path:
+                self.enterprise = self._find_katana_remotes(katana_enterprise_repo_path)
         except CommandError:
             if not self.version_from_environment_variable:
                 logger.warning(
-                    f"Failed to find git repository at {self.katana_repo_path}. "
+                    f"Failed to find git repository at {katana_repo_path}. "
                     f"Version information will be very limited."
                 )
+        except ConfigurationError as e:
+            if not self.version_from_environment_variable:
+                logger.warning(
+                    f"Failed to determine the repo structure. Ignoring git information. "
+                    f"Version information will be very limited.\n{e}",
+                )
+
+    @staticmethod
+    def _find_katana_remotes(dir):
+        name = Path(dir).name
+        remotes = set(git.get_remotes(dir))
+        upstream_remote = None
+        origin_remote = None
+        for remote in remotes:
+            # if we find a remote with an obvious name, override anything we already found
+            if remote in {"upstream"}:
+                upstream_remote = remote
+            elif remote in {"origin"}:
+                origin_remote = remote
+            # If the remote URL has user "KatanaGraph" then it's our upstream (as long as we have not already found an
+            # upstream.
+            url = git.get_remote_url(remote, dir)
+            if not upstream_remote and url.username == UPSTREAM_USERNAME:
+                upstream_remote = remote
+
```

Comment:
is it possible for `upstream_remote` to be `None` after this loop?

Reply:
Yes. Good catch. I'm updating this to fallback on `origin_remote` as `upstream_remote`. And I'm adding the remote names to the logging so users can see which remotes are selected with an appropraite `--log`.